packages/handlersjs-http/lib/handlers/error.handler.ts

```diff
@@ -1,89 +1,90 @@
-      case 422: return of(this.createErrorResponse(response, 'Unprocessable Entity', this.showUpstreamError));
-      case 423: return of(this.createErrorResponse(response, 'Locked', this.showUpstreamError));
-      case 424: return of(this.createErrorResponse(response, 'Failed Dependency', this.showUpstreamError));
-      case 425: return of(this.createErrorResponse(response, 'Too Early', this.showUpstreamError));
-      case 426: return of(this.createErrorResponse(response, 'Upgrade Required', this.showUpstreamError));
-      case 428: return of(this.createErrorResponse(response, 'Precondition required', this.showUpstreamError));
-      case 429: return of(this.createErrorResponse(response, 'Too Many Requests', this.showUpstreamError));
-      case 431: return of(this.createErrorResponse(response, 'Request Header Fields Too Large', this.showUpstreamError));
-      case 451: return of(this.createErrorResponse(response, 'Unavailable For Legal Reasons', this.showUpstreamError));
-      case 500: return of(this.createErrorResponse(response, 'Internal Server Error', this.showUpstreamError));
-      case 501: return of(this.createErrorResponse(response, 'Not Implemented', this.showUpstreamError));
-      case 502: return of(this.createErrorResponse(response, 'Bad Gateway', this.showUpstreamError));
-      case 503: return of(this.createErrorResponse(response, 'Service Unavailable', this.showUpstreamError));
-      case 504: return of(this.createErrorResponse(response, 'Gateway Timeout', this.showUpstreamError));
-      case 505: return of(this.createErrorResponse(response, 'HTTP Version Not Supported', this.showUpstreamError));
-      case 506: return of(this.createErrorResponse(response, 'Variant Also Negotiates', this.showUpstreamError));
-      case 507: return of(this.createErrorResponse(response, 'Insufficient Storage', this.showUpstreamError));
-      case 508: return of(this.createErrorResponse(response, 'Loop detected', this.showUpstreamError));
-      case 510: return of(this.createErrorResponse(response, 'Not Extended', this.showUpstreamError));
-      case 511: return of(this.createErrorResponse(response, 'Network Authentication Required', this.showUpstreamError));
-      default: return response.status < 600 && response.status >= 400
-        ? of(this.createErrorResponse(response, 'An Unexpected Error Occurred', this.showUpstreamError))
-        : of(response);
+      throw new Error('A HttpHandler must be provided');
 
     }
 
   }
 
-  private createErrorResponse(res: HttpHandlerResponse, msg: string, showError: boolean) {
+  handle(context: HttpHandlerContext): Observable<HttpHandlerResponse>{
+
+    if (!context) { return throwError(new Error('A context must be provided')); }
 
-    return showError ? { ...res, body: msg + ': ' + res.body } : { ...res, body: msg };
+    return this.nestedHandler.handle(context).pipe(catchError((error) => of({
```

Comment:
We expect an error response, right? Not an error object? So I assume this should be a switchMap instead of a catchError?